### PR TITLE
chore: use gulp-cache in CI

### PR DESCRIPTION
**Pull request by layershifter**
This PR adds the generated cache by `gulp-cache` to CircleCI cache so it will be possible speed up unit tests and Screener stages.

See #661 for details about `gulp-cache`.

### CODE REVIEW

**Comment by kenotron** [comment]
Cool change! I wonder how much gains you are expecting with this change?

@ecraig12345 is looking at helping make the /docs into a full package. This kind of optimization should be preserved

**Comment by layershifter** [comment]
@kenotron I don't think it's really works 😭 I am going to spend more time tomorrow to make it working. I expect that it should save us around 3 mins on unit tests & Screener:

```
[07:24:14] Finished 'build:docs:component-menu-behaviors' after 2.32 min
[07:24:51] Finished 'build:docs:component-info' after 2.92 min
// Time spent there
```

**Comment by layershifter** [comment]
image

Now it works 🚀

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,6 +38,19 @@ jobs:
             - ~/.cache/yarn
             - .yarn-cache
             - node_modules
+
+      - restore_cache:
+          keys:
+            - v1-json-{{ .Environment.CIRCLE_SHA1 }}
```
Comment (miroslavstastny) [inline .circleci/config.yml]: This is great for rerunning a build on the same commit which does not happen that often.
We will see how useful the cache is with different builds in different branches.

**Review by miroslavstastny — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (layershifter) [reply .circleci/config.yml]: Oops, good catch, should include `CIRCLE_BRANCH`, so there will be more cache hits

**Pull request merged**